controllers/cloud.redhat.com/clowdjobinvocation_controller.go

```diff
@@ -152,125 +161,280 @@ func (r *ClowdJobInvocationReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
+		Image:           fmt.Sprintf("%s:%s", iqeImage, tag),
+		Command:         constructedIqeCommand,
+		Env:             envvar,
+		Resources:       deployProvider.ProcessResources(&pod, env),
+		VolumeMounts:    []core.VolumeMount{},
 		ImagePullPolicy: core.PullIfNotPresent,
 	}
 
-	if (core.Probe{}) != livenessProbe {
-		c.LivenessProbe = &livenessProbe
-	}
-	if (core.Probe{}) != readinessProbe {
-		c.ReadinessProbe = &readinessProbe
-	}
+	j.Spec.Template.Spec.Volumes = []core.Volume{}
+	configAccess := env.Spec.Providers.Iqe.ConfigAccess
+
+	switch configAccess {
+	// Build cdenvconfig.json and mount it
+	case "environment":
+		if secretErr := r.createAndApplyIqeSecret(cache, ctx, cji, env.Name); secretErr != nil {
+			r.Log.Error(secretErr, "Cannot apply iqe secret")
+		}
+		c.VolumeMounts = append(c.VolumeMounts, core.VolumeMount{
+			Name:      "cdenvconfig",
+			MountPath: "/cdenv",
+		})
+
+		j.Spec.Template.Spec.Volumes = append(j.Spec.Template.Spec.Volumes, core.Volume{
+			Name: "cdenvconfig",
+			VolumeSource: core.VolumeSource{
+				Secret: &core.SecretVolumeSource{
+					SecretName: fmt.Sprintf("%s-iqe", cji.Spec.AppName),
+				},
+			},
+		})
+		// if we have env access, we also want app access as well, so also run the next case
+		fallthrough
+
+	// mount cdappconfig
+	case "app":
+		c.VolumeMounts = append(c.VolumeMounts, core.VolumeMount{
+			Name:      "config-secret",
+			MountPath: "/cdapp",
+		})
+
+		j.Spec.Template.Spec.Volumes = append(j.Spec.Template.Spec.Volumes, core.Volume{
+			Name: "config-secret",
+			VolumeSource: core.VolumeSource{
+				Secret: &core.SecretVolumeSource{
+					SecretName: cji.Spec.AppName,
+				},
+			},
+		})
 
-	c.VolumeMounts = append(c.VolumeMounts, core.VolumeMount{
-		Name:      "config-secret",
-		MountPath: "/cdapp/",
-	})
+	case "none":
+		r.Log.Info("No config mounted to the iqe pod")
+	}
+	// c.VolumeMounts = append(c.VolumeMounts, core.VolumeMount{
+	// 	Name:      "iqe-settings",
+	// 	MountPath: "/iqe_settings",
+	// })
+
+	// j.Spec.Template.Spec.Volumes = append(j.Spec.Template.Spec.Volumes, core.Volume{
+	// 	Name: "iqe-settings",
+	// 	VolumeSource: core.VolumeSource{
+	// 		Secret: &core.SecretVolumeSource{
+	// 			SecretName: "iqe-settings",
+	// 		},
+	// 	},
+	// })
 
 	j.Spec.Template.Spec.Containers = []core.Container{c}
 
-	j.Spec.Template.Spec.InitContainers = deployProvider.ProcessInitContainers(nn, &c, pod.InitContainers)
+}
 
-	j.Spec.Template.Spec.Volumes = pod.Volumes
-	j.Spec.Template.Spec.Volumes = append(j.Spec.Template.Spec.Volumes, core.Volume{
-		Name: "config-secret",
-		VolumeSource: core.VolumeSource{
-			Secret: &core.SecretVolumeSource{
-				SecretName: cji.Spec.AppName,
-			},
-		},
-	})
+func constructIqeCommand(cji *crd.ClowdJobInvocation, plugin string) []string {
+	command := []string{
+		"iqe", "tests", "plugin",
+		fmt.Sprintf("%v", strings.ReplaceAll(plugin, "-", "_")),
+		"-m", cji.Spec.Iqe.Marker,
```

Comment:
Still gotta account for these being null....

Reply:
Marker can't be null via the spec, but I added a check for the filter. Also seems like passing `-k ""` does nothing in the iqe plugins.

Reply:
Yeh thanks for the check on that, should we do the same for Marker?